agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -234,245 +134,318 @@ public CallReplacementTransformer() {}
+      return null;
+    }
+
+    debug_transform.log(
+        "%s.%s: Replacing method %s with %s%n",
+        mg.getClassName(), mg.getName(), orig.toString(), call.getQualifiedName());
+
+    InstructionList instructionList = new InstructionList();
+
+    switch (inst.getOpcode()) {
+      case Const.INVOKEDYNAMIC:
+      case Const.INVOKEINTERFACE:
+      case Const.INVOKESPECIAL:
       case Const.INVOKEVIRTUAL:
-        {
-          InstructionList il = new InstructionList();
-          INVOKEVIRTUAL iv = (INVOKEVIRTUAL) inst;
-          String cname = iv.getClassName(pgen);
-          String mname = iv.getMethodName(pgen);
-          Type[] args = iv.getArgumentTypes(pgen);
-          Type instance_type = iv.getReferenceType(pgen);
-          Type[] new_args = BCELUtil.insert_type(instance_type, args);
-          MethodDef orig = new MethodDef(cname + "." + mname, args);
-          if (debug_class)
-            System.out.printf("looking for %s in map %s%n", orig, current_method_map);
-          ReplacementClass call = current_method_map.get(orig);
-          if (call != null) {
-            call.cnt++;
-            String classname = call.method_class;
-            String methodname = mname;
-            debug_map.log(
-                "Replacing method %s.%s (%s) with %s.%s%n",
-                cname, mname, ArraysMDE.toString(args), classname, methodname);
-            il.append(
-                ifact.createInvoke(
-                    classname, methodname, iv.getReturnType(pgen), new_args, Const.INVOKESTATIC));
-          }
-          return (il);
-        }
+        /*
+         * These are invocations where an implicit argument occurs in the call.
+         */
+        Type instanceType = invocation.getReferenceType(pgen);
+        @UnknownInterned
+        @UnknownKeyFor
+        @NonNull
+        @Initialized
+        Type[] arguments = BCELUtil.insert_type(instanceType, invocation.getArgumentTypes(pgen));
+        instructionList.append(
+            ifact.createInvoke(
+                call.getClassname(),
+                call.getName(),
+                invocation.getReturnType(pgen),
+                arguments,
+                Const.INVOKESTATIC));
+        break;
+
+      case Const.INVOKESTATIC:
+        instructionList.append(
+            ifact.createInvoke(
+                call.getClassname(),
+                call.getName(),
+                invocation.getReturnType(pgen),
+                invocation.getArgumentTypes(pgen),
+                Const.INVOKESTATIC));
+        break;
 
       default:
-        return (null);
+        return null;
+    }
+    return instructionList;
+  }
+
+  /**
+   * Returns the replacement method for the given method if one is determined by a method, class or
+   * package replacement, or null otherwise.
+   *
+   * <p>Class or package replacements are represented as strings in the {@link
+   * #prefixReplacementMap}. When the argument method belongs to one of these classes/packages, a
```

Comment:
I find this comment confusing, perhaps in part because the fields and their relationship has not been documented.